frontend/src/components/scheduleEntry/FormScheduleEntryItem.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+  <v-row
+    no-gutters class="mx-2 mb-2">
+    <v-col cols="5">
+      <e-date-picker
+        v-model="mappedScheduleEntry.startTimeUTCFormatted"
+        value-format="YYYY-MM-DDTHH:mm:ssZ"
+        :name="$tc('components.activity.createScheduleEntries.fields.startTime')"
+        vee-rules="required"
+        :allowed-dates="allowedStartDates"
+        :filled="false"
+        class="float-left date-picker"
+        required />
+
+      <e-time-picker
+        v-model="mappedScheduleEntry.startTimeUTCFormatted"
+        :name="$tc('components.activity.createScheduleEntries.fields.startTime')"
+        vee-rules="required"
+        :filled="false"
+        class="float-left mt-0 ml-1 time-picker"
+        required />
+    </v-col>
+    <v-col cols="1" class="text-center pt-4">
+      -
+    </v-col>
+    <v-col cols="5">
+      <e-date-picker
+        v-model="mappedScheduleEntry.endTimeUTCFormatted"
+        value-format="YYYY-MM-DDTHH:mm:ssZ"
+        :name="$tc('components.activity.createScheduleEntries.fields.endTime')"
+        vee-rules="required"
+        :allowed-dates="allowedEndDates"
+        :filled="false"
+        class="float-left date-picker"
+        required />
+
+      <e-time-picker
+        v-model="mappedScheduleEntry.endTimeUTCFormatted"
+
+        :name="$tc('components.activity.createScheduleEntries.fields.endTime')"
+        vee-rules="required"
+        :filled="false"
+        class="float-left mt-0 ml-1 time-picker"
+        required />
+    </v-col>
+
+    <v-col cols="1" class="pt-3 text-center">
+      <button-delete v-if="!isLastItem" icon-only @click="$emit('delete')" />
+    </v-col>
+  </v-row>
+</template>
+<script>
+import { defineHelpers } from '@/common/helpers/scheduleEntry/dateHelperUTCFormatted.js'
+import dayjs from '@/common/helpers/dayjs.js'
+
+import ButtonDelete from '@/components/buttons/ButtonDelete.vue'
+
+export default {
+  name: 'CreateActivityScheduleEntriesItem',
+  components: { ButtonDelete },
+  props: {
+    // scheduleEntry to display
+    scheduleEntry: {
+      type: Object,
+      required: true
+    },
+
+    // List of available periods
+    periods: {
+      type: Array,
+      required: true
+    },
+
+    // true if current item is the last scheduleEntry
+    isLastItem: {
+      type: Boolean,
+      required: true
+    }
+  },
+  data () {
+    return {
+      localScheduleEntry: this.scheduleEntry
+    }
+  },
+  computed: {
+    mappedScheduleEntry () {
+      return defineHelpers(this.localScheduleEntry)
+    },
+
+    // detect selected period based on start date
+    period () {
+      const startDate = dayjs.utc(this.mappedScheduleEntry.startTimeUTCFormatted)
+
+      return this.periods.find((period) => {
+        return startDate.isBetween(dayjs.utc(period.start), dayjs.utc(period.end), 'date', '[]')
+      })
+    }
+  },
+  watch: {
+    'period._meta.self': function () {
```

Comment:
Wie kommt der Watcher zurecht wenn `period` mal nicht existiert? Sollte normalerweise ja nicht vorkommen, aber es könnte ja jemand inzwischen die Periode gelöscht haben und irgendwo sonst Code gerade die Perioden neu laden und somit `period` aus dem Store löschen. Bräuchte es im `computed period` ein Fallback auf `this.periods.find(() => true)`?

Reply:
Good point. Kann man im Moment noch easy getriggert werden, wenn man ein beliebiges Startdatum wählt, welches nicht in einer Period liegt (nicht via Picker sondern via Textfeld).

Resultat:
- Watcher kommt damit ganz gut zurecht, wird einfach mit Wert `undefined` aufgerufen
- Der Code im Watcher kommt damit aber nicht zurecht

Fixed mit commit https://github.com/ecamp/ecamp3/pull/2366/commits/5c3c1039952193e3e46e3b3cdd12d59339038fea. Weiteres kommt dann mit separatem PR für verbessertes Backend+Frontend validation (mach ich, wenn generelles Look & Feel von diesem PR akzeptiert ist).